homeassistant/components/generic_thermostat/climate.py

```diff
@@ -177,11 +205,35 @@ def __init__(
         self._max_temp = max_temp
         self._target_temp = target_temp
         self._unit = unit
-        self._support_flags = SUPPORT_FLAGS
+
+        self._presets = collections.OrderedDict()
```

Comment:
This is not necessary. Starting Python 3.6 all dictionaries have been ordered.
```suggestion
        self._presets = {}
```